mis_funciones.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+* @file mis_funciones.cpp
+* @version 1.0
+* @date 18/7/24
+* @author Ramos Leonardo
+*/
+#include "mis_funciones.hpp"
+
+/// @cond VARIABLES EXTERNAS
+extern BusOut keypadRows;                         
+extern BusIn keypadCols;                         
+extern DigitalOut workingLed;                     
+extern DigitalOut accessGrantedLed;              
+extern DigitalOut accessDeniedLed;               
+extern UnbufferedSerial uartUSB;                 
+
+extern bool sequenceDetectedState;                      
+extern bool accessState;                                
+extern bool timeoutState;                               
+extern int colReading;                                
+extern int rowReading;                                  
+extern uint8_t currentIndex;                            
+extern uint16_t accumulatedTime;                        
+extern char inputSequence[SEQUENCE_LEN + 1];            
+extern char keypadAllowedEntries[ROW_LEN][COL_LEN];    
+/// @endcond
+
+/**
+ * @brief keypadInit Inicializa el keypad 4x4 para su uso. Principalmente
+ *        configura los GPIO de columnas como input PULL DOWN y da valores iniciales
+ *        a ciertos flags. 
+ */
+void keypadInit() {
+  keypadCols.mode(PullDown);
+  inputSequence[SEQUENCE_LEN] = '\0';
+  sequenceDetectedState = false;
+  currentIndex = 0;
+  accumulatedTime = 0;
+}
+
+/**
+ * @brief uartInit Configura velocidad de transmisión y formato de la trama
+ *        de datos de UART
+ */
+void uartInit() {
+  uartUSB.baud(9600);
+  uartUSB.format(/*bits*/ 8,
+                 /*paridad*/ SerialBase::None,
+                 /*bits de stop*/ 1);
+}
+
+/**
+ * @brief keypadSweepUpdate Barre el keypad 4x4 hasta encontrar
+ *        una tecla presionada, guarda el caracter asociado
+ *        a esa tecla e indica mediante flag sequenceDetectedState
+ *        que se ha ingresado la secuencia completa
+ */
+void keypadSweepUpdate() {
+  for (uint8_t i = 0; i < ROW_LEN; i++) {
+    keypadRows = (1 << i);
```

Comment:
Esto en Arm no es mayor problema; en otros micros significa montones de rotaciones inútiles en cada iteración.

Reply:
Gracias por el comentario, quizá deba explorar otra forma de resolverlo.